rhea/cores/fifo/fifo_fast.py

```diff
@@ -97,7 +97,10 @@ def rtl_fifo():
 
         elif fbus.write and not fbus.read:
             fbus.empty.next = False
-            if not fbus.empty:
+            # because addr starts from 0,
+            # there should not be an increment
+            # the first time 
+            if (not fbus.empty) and (not fbus.full):
```

Comment:
I am confused, I don't understand how the previous tests would have passed?  Something is not lining up, either the previous tests were invalid or this change is incorrect.